src/java/org/apache/cassandra/utils/memory/MemtableCleanerThread.java

```diff
@@ -27,54 +33,80 @@
  */
 public class MemtableCleanerThread<P extends MemtablePool> extends InfiniteLoopExecutor
 {
-    private static class Clean<P extends MemtablePool> implements InterruptibleRunnable
+    private static final Logger logger = LoggerFactory.getLogger(MemtableCleanerThread.class);
+    public static class Clean<P extends MemtablePool> implements InterruptibleRunnable
     {
+        /** This is incremented when a cleaner is invoked and decremented when a cleaner has completed */
+        final static AtomicInteger numPendingTasks = new AtomicInteger(0);
```

Comment:
There are a couple things I'd suggest around all this that sort of reinforce each other, so I'll try to list them all here instead of inline.

1.) The `MemtableCleanerThread` is already a singleton, and it might make sense not to make `numPendingTasks` static. We can access it from the `MCT` pretty easily if we just keep a reference to the `Clean` instance.

2.) Having to break encapsulation and reach into `Clean` to get the metric seems suboptimal when we could access it through `MCT`. (The `numPendingTasks()` method in `MCT` is unused, it seems...)

3.) Accessing the `MCT` after that for metric population shouldn't be that hard. We could get it from `Memtable.MEMORY_POOL` or from `DD` itself...